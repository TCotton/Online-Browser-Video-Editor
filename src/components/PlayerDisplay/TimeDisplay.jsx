import React from 'react';
import {useSelector} from "react-redux";
import {time} from '../slices/videoSlice';
import {toMinutesAndSeconds} from "../helperFunctions";

const DurationDisplay = () => {
    const timeDisplay = useSelector(time);

    return (
        <time data-testid="time">{toMinutesAndSeconds(timeDisplay)}</time>
    )
}

export default DurationDisplay
