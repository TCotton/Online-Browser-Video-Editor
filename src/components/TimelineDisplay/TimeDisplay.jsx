import React, {useRef, useEffect, useState} from 'react';
import {useSelector} from "react-redux";
import {window} from 'browser-monads';
import {duration, time} from '../slices/videoSlice';
import Counter from '../../../static/counter.inline.svg'

export const TimeDisplay = () => {
    const [translate, setTranslate] = useState(-17);
    const container = useRef();
    const durationDisplay = useSelector(time);
    const timeDisplay = useSelector(duration);

    useEffect(() => {
        const domRect = container.current.getBoundingClientRect();
        const perSecond = domRect.width / timeDisplay;
        const currentTime = durationDisplay;

        console.log(currentTime, perSecond);

        if(window.Number.isFinite(currentTime) && window.Number.isFinite(perSecond)) {
            setTranslate(() => {
                return (currentTime * perSecond) + (-17);
            });
        }

    },[container, durationDisplay, timeDisplay]);

    return (
        <div ref={container} className="timeline-display" data-testid='TimeDisplay'>
            <Counter style={{ transform: `translate(${translate}px, -230px)` }} />
        </div>
    )
}