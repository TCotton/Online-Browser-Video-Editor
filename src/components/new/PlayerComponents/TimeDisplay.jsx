import React from 'react';
import {useSelector} from "react-redux";
import {time} from '../slices/videoSlice';

const TimeDisplay = () => {
    const timeDisplay = useSelector(time);

    return (
        <time>{timeDisplay.toFixed(3)}</time>
    )
}

export default TimeDisplay
