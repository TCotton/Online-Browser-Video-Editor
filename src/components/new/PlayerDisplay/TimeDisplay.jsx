import React from 'react';
import {useSelector} from "react-redux";
import {time} from '../slices/videoSlice';

const TimeDisplay = () => {
    const timeDisplay = useSelector(time);

    return (
        <time data-testid="time">{timeDisplay.toFixed(3)}</time>
    )
}

export default TimeDisplay
