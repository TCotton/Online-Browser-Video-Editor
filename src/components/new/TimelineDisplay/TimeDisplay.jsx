import React, {useRef, useEffect, useState} from 'react';
import {useSelector} from "react-redux";
import {duration, time} from '../slices/videoSlice';
import Counter from '../../../../static/counter.inline.svg'

export const TimeDisplay = () => {
    const [translate, setTranslate] = useState(-17);
    const container = useRef();
    const durationDisplay = useSelector(duration);
    const timeDisplay = useSelector(time);

    useEffect(() => {
        const domRect = container.current.getBoundingClientRect();
        const perSecond = domRect.width / timeDisplay;
        const currentTime = durationDisplay;

        if(Number.isFinite(currentTime) && Number.isFinite(perSecond)) {
            setTranslate(() => {
                return (currentTime * perSecond) + (-17);
            });
        }


    },[container, durationDisplay, timeDisplay]);

    return (
        <div ref={container} className="timeline-display">
            <Counter style={{ transform: `translate(${translate}px, -230px)` }} />
        </div>
    )
}