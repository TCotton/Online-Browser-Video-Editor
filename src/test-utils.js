import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

import counterReducer from './components/slices/counterSlice';
import filesReducer from './components/slices/filesSlice';
import videoReducer from './components/slices/videoSlice';
import playerReducer from './components/slices/playerSlice';
import audioReducer from './components/slices/audioSlice';
import imageReducer from './components/slices/imageSlice';
import waveformReducer from './components/slices/waveformSlice';

function render(
    ui,
    {
        initialState,
        store = createStore(combineReducers({
            counter: counterReducer,
            files:  filesReducer,
            video: videoReducer,
            player: playerReducer,
            audio: audioReducer,
            images: imageReducer,
            waveform: waveformReducer,
        }), initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react'
// override render method
export { render }