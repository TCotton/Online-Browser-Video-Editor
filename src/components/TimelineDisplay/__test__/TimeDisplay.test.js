import React from 'react';
//import {render} from "@testing-library/react";
import {TimeDisplay} from "../TimeDisplay";
import {render} from '../../../test-utils'

describe('TimeDisplay', () => {

    it('should render successfully', () => {
        const {container} = render(<TimeDisplay/>, {
            initialState: {
                video: {
                    duration: 999,
                    time: 999,
                }
            }
        });
        expect(container.querySelector("[data-testid=\"TimeDisplay\"]")).not.toBeNull();
    });

    it('should match snapshot', () => {
        const {baseElement} = render(<TimeDisplay/>, {
            initialState: {
                video: {
                    duration: 999,
                    time: 999,
                }
            }
        });
        expect(baseElement).toMatchSnapshot();
    });

});