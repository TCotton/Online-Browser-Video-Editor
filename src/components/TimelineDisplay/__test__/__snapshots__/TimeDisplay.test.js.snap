// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeDisplay should match snapshot 1`] = `
<body>
  <div>
    <div
      class="timeline-display"
      data-testid="TimeDisplay"
    >
      <test-file-stub
        style="transform: translate(-17px, -230px);"
      />
    </div>
  </div>
</body>
`;
